tls/s2n_connection.c

```diff
@@ -175,7 +177,8 @@ static int s2n_connection_free_keys(struct s2n_connection *conn)
     return 0;
 }
 
-static int s2n_connection_zero(struct s2n_connection *conn, int mode, struct s2n_config *config)
+static int s2n_connection_zero(struct s2n_connection *conn, int mode, struct s2n_config *config,
+                               struct s2n_signed_evp_digest p_hash_evp_hmac, const struct s2n_p_hash_implementation *p_hash_impl)
```

Comment:
Let's use a pointer for this to avoid the copies. `struct s2n_signed_evp_digest *p_hash_evp_hmac`.